tests/test_codecs.py

```diff
@@ -453,6 +456,51 @@ def test_round_trip(tracer, fmt, carrier):
         assert span.trace_id == span2.trace_id
 
 
+def _text_codec_to_trace_id_string(carrier):
+    return carrier[constants.TRACE_ID_HEADER].split(':')[0]
+
+
+def _zipkin_codec_to_trace_id_string(carrier):
+    return '{:x}'.format(carrier['trace_id'])
+
+
+@pytest.mark.parametrize('fmt,carrier,get_trace_id', [
+    (Format.TEXT_MAP, {}, _text_codec_to_trace_id_string),
+    (Format.HTTP_HEADERS, {}, _text_codec_to_trace_id_string),
+    (ZipkinSpanFormat, {}, _zipkin_codec_to_trace_id_string),
+])
+def test_inject_with_128bit_trace_id(tracer, fmt, carrier, get_trace_id):
+    tracer_128bit = Tracer(
+        service_name='test',
+        reporter=InMemoryReporter(),
+        sampler=ConstSampler(True),
+        generate_128bit_trace_id=True)
+
+    for tracer, length in [(tracer, 16), (tracer_128bit, 32)]:
+        trace_id = (1 << 64) - 1 if length == 16 else (1 << 128) - 1
+        ctx = SpanContext(trace_id=trace_id, span_id=127, parent_id=None,
+                          flags=1)
+        span = Span(ctx, operation_name='test-%s' % fmt, tracer=None, start_time=1)
```

Comment:
Could you use extract from a string instead of creating the span context manually? The idea was to test passing trace id unchanged end-to-end

Reply:
Umm, the end to end trace_id is being already tested by this modified round_trip here https://github.com/jaegertracing/jaeger-client-python/pull/230/files#diff-252cc4202700183e8f0c782fb28dbfc2R451

Reply:
It's not testing the hybrid mode, but perhaps we can do it later in another PR if the need arrises. AFAIK other Jaeger clients that support 128bit ID also only support a full switch mode.